docs/quickstart.qbk

```diff
@@ -0,0 +1,282 @@
+`simplest_hello_world_1` example by running
+
+``
+    make simplest_hello_world_1_exe
+``
+
+__hpx__ executables end up in the `bin` directory in your build directory. You
+can now run `simplest_hello_world` and should see the following output:
+
+``
+./bin/simplest_hello_world_1
+Hello World!
+``
+
+You've just run an example which prints `Hello World!` from the __hpx__ runtime.
+The source for the example is in `examples/quickstart/simplest_hello_world.cpp`.
+The `hello_world` example (also available in the `examples/quickstart` directory
+is a distributed hello world program which is described in
+__hello_world_example__. It is recommended that you go through that example only
+once you are more familiar with the distributed concepts of __HPX__.
+
+[tip Most build targets in __hpx__ have two names: a simple name suffixed with
+`_exe` and a hierarchical name corresponding to what type of example or test the
+target is. If you are developing __hpx__ it is often helpful to run `make help`
+to get a list of available targets. For example, `make help | grep hello_world`
+outputs the following:
+
+``
+... examples.quickstart.simplest_hello_world_2
+... simplest_hello_world_2_exe
+... examples.quickstart.simplest_hello_world_1
+... simplest_hello_world_1_exe
+... hello_world_exe
+... examples.quickstart.hello_world
+``
+
+It is also possible to build e.g. all quickstart examples using `make
+examples.quickstart`. ]
+
+[heading Setting up an HPX project]
+
+The following `CMakeLists.txt` is a minimal example of what you need in order to
+build an executable using __cmake__ and __hpx__:
+
+``
+cmake_minimum_required(VERSION 3.5)
+project(my_hpx_project CXX)
+find_package(hpx REQUIRED)
+add_hpx_executable(my_hpx_program SOURCES main.cpp)
+``
+
+[note You will most likely have more than one `main.cpp` file in your project.
+See the section on __using_hpx_cmake__ for more details on how to use
+`add_hpx_executable`. ]
+
+Create a new project directory and a `CMakeLists.txt` with the contents above.
+Then, in your project directory run the following:
+
+``
+touch main.cpp # Create an empty file for now
+mkdir build && cd build
+cmake -DCMAKE_PREFIX_PATH=/path/to/hpx/installation ..
+make all
+./my_hpx_program
+``
+
+Your program should now start but fail quickly because it doesn't contain a main
+function yet.
+
+[heading Hello, World!]
+
+Once you have your project set up, add the following to your `main.cpp`:
+
+[import ../examples/quickstart/simplest_hello_world_1.cpp]
+
+[simplest_hello_world_1_getting_started]
+
+It looks almost like a regular `C++` hello world with the exception of the two
+includes and `hpx::cout`. When you include `hpx_main.hpp` some things will be
+done behind the scenes to make sure that `main` actually get's launched on the
+__hpx__ runtime. So while it looks almost the same you can now use futures,
+`async`, parallel algorithms and more which make use of the __hpx__ runtime with
```

Comment:
The only difference to a 'normal' C++ program is that your `main` function _must_ return a value (normally main is allowed not to have an explicit `return` statement).

Reply:
Very good point. We already got bitten by this at CSCS, so it would be very good to point that out. I'll add a note.